src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -323,40 +307,30 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
     /**
      *  Given a FedoraResource, return a list of agents.
      */
-    private Set<String> getAgentMembers(final IdentifierConverter<Resource, FedoraResource> translator,
+    private static Stream<String> getAgentMembers(final IdentifierConverter<Resource, FedoraResource> translator,
             final FedoraResource resource) {
-        final Set<String> members = new HashSet<>();
-        final Model model = createDefaultModel();
-
-        final Predicate<Property> isMember = memberTestFromTypes.apply(resource.getTypes());
-
-        resource.getTriples(translator, PROPERTIES)
-            .filter(p -> isMember.test(model.asStatement(p).getPredicate()))
-            .forEach(t -> {
-                if (t.getObject().isURI()) {
-                    members.add(t.getObject().getURI());
-                } else if (t.getObject().isLiteral()) {
-                    members.add(t.getObject().getLiteralValue().toString());
-                }
-            });
-
-        return members;
+        return resource.getTriples(translator, PROPERTIES).filter(memberTestFromTypes.apply(resource.getTypes()))
+            .map(Triple::getObject).flatMap(nodeToStringStream);
     }
 
     /**
+     * Map a Jena Node to a Stream of Strings. Any non-URI, non-Literals map to an empty Stream,
+     * making this suitable to use with flatMap.
+     */
+    private static final Function<com.hp.hpl.jena.graph.Node, Stream<String>> nodeToStringStream = object ->
```

Comment:
I would just let this be a method returning a `Stream` here, not a function. Then above you can use `WebACRolesProvider::nodeToStringStream` and below you can remove the `apply`. That would seem to me to be clearer, but maybe it's a matter of taste.